cms/djangoapps/contentstore/views/block.py

```diff
@@ -1405,6 +1407,11 @@ def create_xblock_info(xblock, data=None, metadata=None, include_ancestor_info=F
             xblock_info['ancestor_has_staff_lock'] = False
 
         if course_outline:
+            if xblock_info['optional_completion']:
+                xblock_info['ancestor_has_optional_completion'] = ancestor_has_optional_completion(xblock, parent_xblock)
```

Comment:
What if the `optional_completion` is set to `False` (default), but the parent XBlock has it set to `True`? In this use case, the field is still active. We should disable this checkbox when `ancestor_has_optional_completion` returns `True`, regardless of the `xblock_info['optional_completion']` value.

Reply:
good point